apache_httpd-formula/tests/test_httpd.py

```diff
@@ -0,0 +1,107 @@
+"""
+Test suite for assessing the apache_httpd formula
+Copyright (C) 2024 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+
+This program is free software: you can redminetribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import pytest
+
+@pytest.mark.parametrize(
+  'pillar, package', [
+    ({}, 'apache2-event'),
+    ({'sysconfig': {'APACHE_MPM': 'prefork'}}, 'apache2-prefork')
+  ],
+  indirect=['pillar']
+)
+@pytest.mark.parametrize('test', [True, False])
+def test_httpd_package(host, package, pillar, salt_apply, test):
+  result = salt_apply
+  assert len(result) > 0
+  output = result[0]
+  state = 'pkg_|-apache_httpd_packages_|-apache_httpd_packages_|-installed'
+  assert state in output
+  assert output[state].get('name') == 'apache_httpd_packages'
+  changes = output[state].get('changes')
+  if test:
+    assert changes == {package: {'new': 'installed', 'old': ''}}
+  else:
+    assert package in changes
+    assert 'apache2-utils' in changes
+    # in non-test runs, "new" will return the freshly installed version, such as 2.4.51-150400.6.14.1
+    assert '2.4' in changes[package]['new']
+  print(output)
+  assert host.package('apache2').is_installed is not test
+  assert host.package(package).is_installed is not test
+
+
+@pytest.mark.parametrize('test', [True, False])
+def test_httpd_config(host, salt_apply, test):
+  result = salt_apply
+  assert len(result) > 0
+  output = result[0]
+  for file, checksum in {
+    'conf.d/remote.conf': '2d4e69a65a3c77743f8504af4ae2415a',
+    'vhosts.d/mysite1.conf': '15edeaf0ea295a192e9aa964b011493f',
+    'vhosts.d/mysite2.conf': 'b11ed81a8d1e8f2c5f48a89e9df8c757',
+    'vhosts.d/status.conf': '41228b88bdb6156773b4e26c39c5d9bc',
+  }.items():
+    if file.startswith('conf.d'):
+      place = 'configs'
+    elif file.startswith('vhosts.d'):
+      place = 'vhosts'
+    file = f'/etc/apache2/{file}'
+    state = f'file_|-apache_httpd_{place}_|-{file}_|-managed'
+    assert state in output
+    assert output[state].get('name') == file
+    changes = output[state].get('changes')
+    if test:
+      assert changes == {'newfile': file}
+      assert output[state]['result'] is None
+    else:
+      assert changes == {'diff': 'New file', 'mode': '0644'}
+      file = host.file(file)
+      assert file.is_file
+      assert file.uid == 0
+      assert file.gid == 0
+      assert file.md5sum == checksum
```

Comment:
Keeping only the checksum saves some space, but if you ever get a different result, debugging it will be interesting[tm]. I'd recommend to include the whole generated files as reference so that you can diff against them.

Reply:
Yes, good idea, I will add them.
It was less about the space, more about not wanting to write a lot of line-by-line assertions or regex matches with new-lines (which are similarly difficult to figure out in assertion failure messages).